public/lessons/db/mongoose.md

```diff
@@ -0,0 +1,106 @@
+# Overview
+When writing to a database, it is often important to ensure the data types/fields are expected.
+
+Enter schema-validation with **Mongoose**. 
+
+[Mongoose](https://mongoosejs.com/) is an object data modeling (ODM) library.
+ 
+Or in their words,
+
+        "A straight-forward, schema-based solution to model your application data"
+
+# Getting Started
+* Mongoose replaces the `mongodb` driver
+* Runs `mongodb` queries under the hood
+
+Make a new directory named `mongoose-example` 
+
+```
+cd mongoose-example
+npm init -y
+npm install mongoose
+```
+# Getting Started, cont.
+Create a file called `mongoose.js`
+In it, write the following code.
+
+```javascript
+const mongoose = require('mongoose')
+/* 
+pass { useNewUrlParser: true, useUnifiedTopology: true } as a second 
+argument to the connect method to avoid deprecation warnings
+*/
+mongoose.connect('mongodb://localhost:27017/example_db') 
+const db = mongoose.connection
+
+db.on('error', console.error.bind(console, 'connection error:'))
+```
```

Comment:
Lets break this down into more slides, and explain what mongoose does in more depth. Then have the 'code-along' parts be split into dedicated Labs slides.